eZ/Bundle/EzPublishCoreBundle/DependencyInjection/Configuration/ConfigResolver.php

```diff
@@ -237,8 +274,39 @@ public function getDefaultScope()
         return $this->defaultScope ?: $this->siteAccess->name;
     }
 
+    /**
+     * @param string $scope The default "scope" aka siteaccess name, as opposed to the self::SCOPE_DEFAULT.
+     */
     public function setDefaultScope($scope)
     {
         $this->defaultScope = $scope;
+
+        // On scope change check if siteaccess has been updated so we can log warnings if there are any
+        if ($this->siteAccess->matchingType !== 'uninitialized') {
+            $this->logTooEarlyLoadedParams();
+        }
+    }
+
+    private function logTooEarlyLoadedParams()
+    {
+        if (empty($this->tooEarlyLoadedList)) {
+            return;
+        }
+
+        $logger = $this->container->get('logger');
+        foreach ($this->tooEarlyLoadedList as $param => $services) {
+            // Ideally we we would want to skip warnings for services that use dynamic settings on setters as that means
+            // paramter will get update on scope change, but we don't have a way to detect that here.
+            $logger->warning(sprintf(
+                'ConfigResolver was used to load parameter "%s" before SiteAccess was loaded by the following services: %s. This should be avoided; '
+                . 'first try to use ConfigResolver lazily in these services instead of "$dynamic_paramter$" injection, '
+                . (PHP_SAPI == 'cli' ? 'if not possible make sure your commands that rely on them are lazy loaded, ' : '')
+                . 'if nothing else helps try to mark service as lazy.',
+                $param,
+                '"' . implode($services, '", "') . '"'
```

Comment:
The message here might need re-phrasing depending on my other requests.

Reply:
Closing thread as PR changed